engine/src/cmd/energetic.cpp

```diff
@@ -230,3 +256,187 @@ float Energetic::warpEnergyData() const
         return ( (float) warpenergy )/( (float) jump.energy );
     return 0.0f;
 }
+
+
+
+// Basically max or current shield x 0.2
+float Energetic::totalShieldEnergyCapacitance()
+{
+    Unit *unit = static_cast<Unit*>(this);
+    DamageableLayer *shield = unit->shield;
+
+    float total_max_shield_value = shield->TotalMaxLayerValue();
+    float total_current_shield_value = shield->TotalLayerValue();
+
+    static float shield_energy_capacitance =
+        GameConfig::GetVariable( "physics", "shield_energy_capacitance", 0.2f );
+    static bool  use_max_shield_value =
+        GameConfig::GetVariable( "physics", "use_max_shield_energy_usage", false );
+
+    return shield_energy_capacitance * (use_max_shield_value ? total_max_shield_value : total_current_shield_value);
+}
+
+
+// The original code was in unit_generic:5476 RegenShields and was simply
+// incomprehensible. After several days, I've written a similar version.
+// However, someone who understands the previous code can refactor this easily
+// or better yet, write plugable consumption models.
+//GAHHH reactor in units of 100MJ, shields in units of VSD=5.4MJ to make 1MJ of shield use 1/shieldenergycap MJ
+void Energetic::ExpendEnergy(const bool player_ship) {
+    // TODO: if we run out of fuel or energy, we die from lack of air
+
+    MaintainShields();
+    ExpendEnergyToRechargeShields();
+    MaintainECM();
+    DecreaseWarpEnergyInWarp();
+
+    RechargeWarpCapacitors(player_ship);
+
+    ExpendFuel();
+}
+
+
+void Energetic::ExpendEnergy(float usage) {
+    energy = std::max(0.0f, energy - usage);
+}
+
+// The original code was a continuation of the comment above and simply unclear.
+// I replaced it with a very simple model.
+void Energetic::ExpendFuel() {
+    //Fuel Mass in metric tons expended per generation of 100MJ
+    static float FMEC_factor = GameConfig::GetVariable( "physics", "FMEC_factor", 0.000000008 );
```

Comment:
Just to not: this is pushing the boundaries of what a float can compute, depending on hardware. Most hardware should be 9 decimal now, but it's possible it could be just 6 or 7 digit. (Not a blocker...just noting it)